web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/CreateCertificateFromTemplateFunctionImpl.java

```diff
@@ -0,0 +1,157 @@
+import se.inera.intyg.common.support.facade.model.Certificate;
+import se.inera.intyg.common.support.facade.model.CertificateRelationType;
+import se.inera.intyg.common.support.facade.model.CertificateStatus;
+import se.inera.intyg.common.support.facade.model.metadata.CertificateRelations;
+import se.inera.intyg.webcert.web.service.user.dto.WebCertUser;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.ResourceLinkDTO;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.ResourceLinkTypeDTO;
+
+@Component
+public class CreateCertificateFromTemplateFunctionImpl implements CreateCertificateFromTemplateFunction {
+
+    private static final String CREATE_AG7804_NAME = "Skapa AG7804";
+
+    private static final String CREATE_AG7804_DESCRIPTION = "Skapar ett intyg till arbetsgivaren utifrån Försäkringskassans intyg.";
+    private static final String CREATE_AG7804_BODY = "<div><div class=\"ic-alert ic-alert--status ic-alert--info\">\n"
+        + "<i class=\"ic-alert__icon ic-info-icon\"></i>\n"
+        + "Kom ihåg att stämma av med patienten om hen vill att du skickar Läkarintyget för sjukpenning till Försäkringskassan. "
+        + "Gör detta i så fall först.</div>"
+        + "<p class='iu-pt-400'>Skapa ett Läkarintyg om arbetsförmåga - arbetsgivaren (AG7804)"
+        + " utifrån ett Läkarintyg för sjukpenning innebär att "
+        + "informationsmängder som är gemensamma för båda intygen automatiskt förifylls.\n"
+        + "</p></div>";
+
+    private static final String CREATE_DOI_NAME = "Skapa dödsorsaksintyg";
+    private static final String CREATE_DOI_DESCRIPTION = "Skapar ett dödsorsaksintyg utifrån dödsbeviset.";
+    private static final String CREATE_DOI_BODY = "Skapa ett dödsorsaksintyg utifrån ett dödsbevis innebär att informationsmängder som är "
+        + "gemensamma för båda intygen, automatiskt förifylls.";
+
+    @Override
+    public Optional<ResourceLinkDTO> get(Certificate certificate, WebCertUser webCertUser) {
+        if (isCreateAg7804FromFk7804(certificate, webCertUser)) {
+            return Optional.of(
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.CREATE_CERTIFICATE_FROM_TEMPLATE,
+                    CREATE_AG7804_NAME,
+                    CREATE_AG7804_DESCRIPTION,
+                    CREATE_AG7804_BODY,
+                    true
+                )
+            );
+        }
+
+        if (isCreateDoiFromDb(certificate, webCertUser)) {
+            return Optional.of(
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.CREATE_CERTIFICATE_FROM_TEMPLATE,
+                    CREATE_DOI_NAME,
+                    CREATE_DOI_DESCRIPTION,
+                    CREATE_DOI_BODY,
+                    true
+                )
+            );
+        }
+
+        return Optional.empty();
+    }
+
+    private boolean isCreateAg7804FromFk7804(Certificate certificate, WebCertUser webCertUser) {
```

Comment:
A description or explanation about this buisness logic would be nice :) 
It is kind of hard to know why the method is looking at things like isDjupintegration or hasBeenComplementedBySignedCertificate to determine if Ag7804 is created from Fk7804.